File: quiz_app_flutter/lib/domain/mapper/dashboard_mapper.dart
import '../../data/model/dashboard/dashboard_dto.dart';
import '../entities/dashboard/dashboard_data.dart';

extension DashboardDtoX on DashboardDto {
  DashboardData toDomain() => DashboardData(total: total, items: items.map((e) => e.toDomain()).toList());
}

extension DashboardItemDtoX on ItemDto {
  DashboardDataItem toDomain() => DashboardDataItem(
    total: total,
    title: title,
    topic: topic,
    description: description,
    sectionId: sectionId,
    sections: sections.map((e) => e.toDomain()).toList(),
  );
}

extension DashboardSectionDtoX on SectionDto {
  DashboardDataSection toDomain() =>
      DashboardDataSection(title: title, description: description, position: position, fileName: fileName);
}

File: quiz_app_kmp/shared/src/commonMain/kotlin/com/tnm/quizappkmp/shared/data/mapper/DashboardMapper.kt
package com.tnm.shared.data.mapper

import com.tnm.shared.data.dto.DashboardDto
import com.tnm.quizappkmp.domain.DashboardData


fun DashboardDto.toDomain() = DashboardData(
    total = total,
    items = items.map { it.toDomain() }
)

fun DashboardDto.ItemDto.toDomain() = DashboardData.Item(
    total = total,
    title = title,
    topic = topic,
    description = description,
    sectionId = sectionId,
    sections = sections.map { it.toDomain() }
)

fun DashboardDto.SectionDto.toDomain() = DashboardData.Section(
    title = title,
    description = description,
    position = position,
    fileName = fileName
)